src/lib.rs

```diff
@@ -250,6 +250,12 @@ impl Time {
     pub fn since_epoch(&self) -> Duration {
         Duration::millis(self.as_millis())
     }
+
+    pub fn since_midnight(&self) -> Duration {
+        let millis_in_day = 86400000;
+        let millis_since_midnight = self.as_millis() % millis_in_day;
+        Duration::millis(millis_since_midnight)
+    }
```

Comment:
I don't think this implementation is leap-year safe, nor is it easy to make it so. I think it might be better to use a library such as `chrono` for this.